### ISSUE TITLE
attribute_statements always triggers a change in okta_app_saml when using preconfigured_app

### ISSUE THREAD

**Issue by dkulchinsky** [okta/terraform-provider-okta#415]
### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

```
❯ terraform -v
Terraform v0.14.7
+ provider registry.terraform.io/hashicorp/null v3.1.0
+ provider registry.terraform.io/oktadeveloper/okta v3.10.1
```

### Affected Resource(s)

- okta_app_saml

### Terraform Configuration Files

```hcl
resource "okta_app_saml" "preconfigured_app" {
  label                    = var.app_name
  preconfigured_app        = var.app_market_name
  app_settings_json        = jsonencode({"baseUrl"= var.app_base_url})

  attribute_statements {
    type         = "GROUP"
    name         = "group"
    filter_type  = "REGEX"
    filter_value = ".*"
  }

  // groups are assigned through okta_app_group_assignment resource
  lifecycle {
    ignore_changes = [groups]
  }
}
```

### Terraform Output

The above always produces the following diff in a plan/apply:
```
  # module.elevation_K8s_preconfigured_saml_apps["elevation-ci-sbx-usc1"].okta_app_saml.preconfigured_app will be updated in-place
  ~ resource "okta_app_saml" "preconfigured_app" {
        id                         = "0oab68hcz4iJtFkKt5d6"
        name                       = "jenkins"
        # (24 unchanged attributes hidden)

      + attribute_statements {
          + filter_type  = "REGEX"
          + filter_value = ".*"
          + name         = "group"
          + namespace    = "urn:oasis:names:tc:SAML:2.0:attrname-format:unspecified"
          + type         = "GROUP"
        }
    }
```

### Expected Behavior

The resource should define the `attribute_statements` once and not show any diff unless the configuration was changed.

### Actual Behavior

Every plan produces the above diff.

### Steps to Reproduce

1. create an `okta_app_saml` with a `preconfigured_app` (we use `Jenkins`)
2. apply the code
3. run plan and see the diff appears

### Important Factoids

Commenting the `attribute_statements` block after initial provisioning of the resource stops the diffs from showing up, but not really a fix.

**Comment by bogdanprodan-okta** [comment]
Hi, @dkulchinsky! Thanks for submitting this issue! I'll investigate the problem and will create a PR to fix it.

**Comment by bogdanprodan-okta** [comment]
Hi, @dkulchinsky! Seems like you have the same case as in #418. Jenkins doesn't support `attribute_statements` by default.
I'm not sure yet how to enable that, but I'm working on it.

**Comment by dkulchinsky** [comment]
thank you! @bogdanprodan-okta 

So this is interesting, when we create the App with Terraform it does set the correct SAML properties:

**EDIT**: as described in a comment below, a team mate indeed configured this bit manually after creating the app with terraform.

![image](https://user-images.githubusercontent.com/61992380/114716823-6ed6d300-9d02-11eb-92c3-3a9035b5a0c8.png)

as opposed to creating the same Jenkins App from Okta UI, results in this:
![image](https://user-images.githubusercontent.com/61992380/114716909-857d2a00-9d02-11eb-8382-ed2871db5d1a.png)

**Comment by dkulchinsky** [comment]
@bogdanprodan-okta I just did a quick test, I manually changed the `group` matcher to `None` and applied our TF code, indeed it didn't change it back to `Regex/.*`.

I checked with my team mates and looks like this was defined manually after creating the application with Terraform.

**Comment by bogdanprodan-okta** [comment]
Hi, @dkulchinsky! Please read the [top comment](https://github.com/okta/terraform-provider-okta/pull/439#issue-622030948) in the linked PR. Thanks!

**Comment by dkulchinsky** [comment]
Thanks @bogdanprodan-okta! When you say 'org' do you mean in Okta somewhere? Also, any updates about publishing new version in terraform registry?

**Comment by bogdanprodan-okta** [comment]
@dkulchinsky, yes, I think it can be done either by superadmin or Okta support.

**Comment by dkulchinsky** [comment]
Thanks for confirming @bogdanprodan-okta! any news on when the new releases will find their way into the Terraform registry? 🙏🏼

**Comment by bogdanprodan-okta** [comment]
@dkulchinsky no updates from Hashicorp so-far

**Comment by liath** [comment]
pssst @swampdragons is this something you could help us out on? :D

**Comment by SwampDragons** [comment]
@Liath The Terraform registry isn't my area at all... have y'all reached out to terraform-registry@hahshicorp.com to ask for help there? I thought registry updates were handled via webhooks, so maybe something's broken.

**Comment by bogdanprodan-okta** [comment]
The thing is, that provider has migrated from `oktadeveloper` to `okta` namespace, and the registry is still using `oktadeveloper`.

**Comment by dkulchinsky** [comment]
Hey @bogdanprodan-okta I just bumped the provider to 3.11.1 and tested this fix, unfortunately getting the same issue 😢

I verified that the Org features mentioned in the linked PR are enabled:
![image](https://user-images.githubusercontent.com/61992380/119747885-a8524080-be61-11eb-8a7d-259538695fc6.png)

but still getting the diff about the `attribute_statements` on every plan after applying.

```
  # module.elevation_K8s_preconfigured_saml_apps["<name>"].okta_app_saml.preconfigured_app will be updated in-place
  ~ resource "okta_app_saml" "preconfigured_app" {
        id                         = "0oab68hcz4iJtFkKt5d6"
        name                       = "jenkins"
        # (24 unchanged attributes hidden)

      + attribute_statements {
          + filter_type  = "REGEX"
          + filter_value = ".*"
          + name         = "group"
          + namespace    = "urn:oasis:names:tc:SAML:2.0:attrname-format:unspecified"
          + type         = "GROUP"
        }
    }
```

**Comment by bogdanprodan-okta** [comment]
Hi @dkulchinsky! Unfortunately, this fix is not present in v3.11.1. This version was released long ago but wasn't present in the registry. Currently, I'm working on v4.0.0-beta which will have some breaking changes (like no `users` and `groups` in the apps, all the deprecated fields/resources will be removed, etc.)

**Comment by dkulchinsky** [comment]
Thanks for the update @bogdanprodan-okta! I guess I'm a bit confused, the fix (#439) was merged to `master` on Apr 23 and 3.11.1 was released 2 days ago, so I assumed it would be part of it but now I see that v3.11.1 tag points to a point prior to this commit.

Any reason this fix can't be included in a minor v3.11.x release? is it tightly coupled with the changes introduced in v4?
